app/jobs/csv_export_job.rb

```diff
@@ -0,0 +1,65 @@
+require 'csv'
+
+class CsvExportJob < ActiveJob::Base
+  queue_as :csv_jobs
+ 
+  def perform(csv_export)
+    ActiveRecord::Base.connection_pool.with_connection do
+      create_export_folder(csv_export)
+      export_task(csv_export)
+      cleanup_downloaded
+    end
+  end
+  
+  private
+  
+  def cleanup_downloaded
+    # TODO delete files with no record associations
+    CsvExport.old.each do |csv_export|
+      csv_export.destroy
+    end
+  end
+  
+  def export_task(csv_export)
+    csv_export.status! :started
+    deploy_csv_export(csv_export)
+    notify_of_creation(csv_export)
+  rescue NoMethodError, IOError, ActiveRecord::ActiveRecordError => e
```

Comment:
I wonder if we should explicitly check for deleted records or include them using `Project.with_deleted { ... }`. This code was a bit confusing to me because in most cases `NoMethodError` is a bug